juju/juju
# Merge 2.9

Merge 2.9

#13693 [JUJU-416] Consistantly use juju/retry to handle retries 8 (worker/upgradesteps/*)
#13705 Ensure we check trace enabled before heavy operations
#13711 [JUJU-585] Increase SA secret bootstrap time out.
#13715 [JUJU-587] Tweak cpu power so test passes
#13719 [JUJU-575] Fix/nw deploy kubeflow
#13686 [JUJU-416] Consistantly use juju/retry to handle retries 6 (provider/gce/*)
#13716 [JUJU-586] Supply OpenStack FIPs to the instance-polle
#13718 Add Darwin support for uploading Microk8s images
#13722 Fixes Juju to handle microk8s not running
#13724 [JUJU-575] Reset in kubeflow dir;
#13723 [JUJU-256] Add microk8s tests for upgrade model actions
#13727 [JUJU-600] Add log diff debug message for assess_model_migration.py
#13717 [JUJU-592] LP1960235 worker does not stop
#13714 Fix multiwatcher data race
#13729 [JUJU-603] Remove client websocket buffer size stipulation
#13731 Added --no-progress argument to juju download.
#13733 Fix manual machine checks when migrating manual models
#13728 [JUJU-394] Improve enable-ha error message
#13721 Move lxd profile watcher from model cache to apiserver layer

Conflicts due to version bump and removal of actions UUID support.
```
# Conflicts:
#       .github/workflows/smoke.yml
#       apiserver/allfacades.go
#       apiserver/facades/client/action/action_test.go
#       apiserver/facades/client/client/client.go
#       go.sum
#       scripts/win-installer/setup.iss
#       snap/snapcraft.yaml
#       state/action_test.go
#       state/operation_test.go
#       version/version.go
#       worker/uniter/runner/context/context_test.go
```

## QA steps

See PRs


[JUJU-416]: https://warthogs.atlassian.net/browse/JUJU-416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-585]: https://warthogs.atlassian.net/browse/JUJU-585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-587]: https://warthogs.atlassian.net/browse/JUJU-587?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-575]: https://warthogs.atlassian.net/browse/JUJU-575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-416]: https://warthogs.atlassian.net/browse/JUJU-416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-586]: https://warthogs.atlassian.net/browse/JUJU-586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-575]: https://warthogs.atlassian.net/browse/JUJU-575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-256]: https://warthogs.atlassian.net/browse/JUJU-256?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-600]: https://warthogs.atlassian.net/browse/JUJU-600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-592]: https://warthogs.atlassian.net/browse/JUJU-592?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-603]: https://warthogs.atlassian.net/browse/JUJU-603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-394]: https://warthogs.atlassian.net/browse/JUJU-394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
